Include/internal/pycore_frame.h

```diff
@@ -94,20 +95,20 @@ static inline void _PyFrame_StackPush(_PyInterpreterFrame *f, PyObject *value) {
 
 void _PyFrame_Copy(_PyInterpreterFrame *src, _PyInterpreterFrame *dest);
 
-/* Consumes reference to func */
+/* Consumes reference to func and locals */
 static inline void
 _PyFrame_InitializeSpecials(
     _PyInterpreterFrame *frame, PyFunctionObject *func,
-    PyObject *locals, int nlocalsplus)
+    PyObject *locals, PyCodeObject *code)
 {
     frame->f_func = func;
-    frame->f_code = (PyCodeObject *)Py_NewRef(func->func_code);
+    frame->f_code = (PyCodeObject *)Py_NewRef(code);
     frame->f_builtins = func->func_builtins;
     frame->f_globals = func->func_globals;
-    frame->f_locals = Py_XNewRef(locals);
```

Comment:
Out of interest, why did you move to incref to out of this function? Does this mean there are some situations where the caller already holds a strong reference that's meant to be transferred?

Reply:
This now consumes a reference to `locals`, so any caller passing in non-NULL needs to allow for that.
Since `locals` is almost always `NULL`, this seems more efficient.

Reply:
Ahh ok, so it removes a branch in most cases too (from the `Py_XNewRef`). Cool.